R/parse-body.R

```diff
@@ -519,6 +519,7 @@ parser_multi <- function() {
     if (!stri_detect_fixed(content_type, "boundary=", case_insensitive = TRUE))
       stop("No boundary found in multipart content-type header: ", content_type)
     boundary <- stri_match_first_regex(content_type, "boundary=([^; ]{2,})", case_insensitive = TRUE)[,2]
+    boundary <- stri_replace_first_regex(boundary, '^"([^"]+)"?', "$1")
```

Comment:
I like your implementation that you did in jeroen/webutils#8. I compared this implementation, your issue suggestion, and the webutils PR. The simple `gsub` command was the fastest, but no guarantees a start and end quote exist. 

So we should stay with the current implementation.

```r
boundary <- "\"----WebKitFormBoundaryMYdShB9nBc32BUhQ\""
bench::mark(
  stri_replace_first_regex(boundary, '^"([^"]+)"$', "$1"),
  gsub("^\"|\"$", "", boundary),
  sub('^"(.*)"$', "\\1", boundary)
)
#> # A tibble: 3 × 13
#>   expression        min median `itr/sec` mem_alloc `gc/sec` n_itr  n_gc total_time result memory     time       gc
#>   <bch:expr>     <bch:> <bch:>     <dbl> <bch:byt>    <dbl> <int> <dbl>   <bch:tm> <list> <list>     <list>     <list>
#> 1 "stri_replace… 3.65µs 3.81µs   253590.        0B      0   10000     0     39.4ms <chr>  <Rprofmem> <bench_tm> <tibble>
#> 2 "gsub(\"^\\\"… 3.36µs 3.61µs   268517.        0B      0   10000     0     37.2ms <chr>  <Rprofmem> <bench_tm> <tibble>
#> 3 "sub(\"^\\\"(… 3.81µs 4.02µs   238138.        0B     23.8  9999     1       42ms <chr>  <Rprofmem> <bench_tm> <tibble>
```

Adjusting the regex to force a quote to exist at the end. (If it is improperly formed, I don't know how it'll be parsed in the first place.)

```suggestion
    # Remove surrounding quotes if they exist
    boundary <- stri_replace_first_regex(boundary, '^"([^"]+)"$', "$1")
```